### PR TITLE
Newsletters: Make Subscriber Reach Sentence Future Tense for Scheduled Post

### LINKED ISSUE #31135: Newsletters panel: update wording in post-publish panel for scheduled posts
### Impacted plugin

Jetpack

### Quick summary

_No response_

### Steps to reproduce

1. Go to Posts > Add New
2. Write a post and save as draft.
3. Set a publishing date to the future
4. Schedule your post	

### A clear and concise description of what you expected to happen.

I would expect a message telling me the post **will be** delivered to my subscribers upon publication	

### What actually happened

The message tells me the post was already delivered.

### Impact

All

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Simple, Atomic, Self-hosted

### Logs or notes

_No response_

### PR DESCRIPTION
**Pull request by romarioraffington**
Fixes https://github.com/Automattic/jetpack/issues/31135

## Proposed changes:

Make newsletter subscribers reach sentence future-tense for scheduled posts

| Before | After |
|--------|--------|
| Screenshot 2023-06-06 at 12 36 49 PM | Screenshot 2023-06-06 at 12 55 40 PM

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Go to Posts > Add New
2. Write a post and save it as a draft.
3. Set a publishing date for the future
4. Schedule your post
5. Notice the Prepublish panel displays the subscribers reach in the future tense

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/subscriptions/settings.js [hunk 1]
```diff
@@ -63,11 +63,15 @@ export function NewsletterNotice( {
...
-		select( editorStore ).isCurrentPostPublished()
-	);
+	const { hasPostBeenPublished, hasPostBeenScheduled } = useSelect( select => {
+		const { isCurrentPostPublished, isCurrentPostScheduled } = select( editorStore );
+
+		return {
+			hasPostBeenPublished: isCurrentPostPublished(),
+			hasPostBeenScheduled: isCurrentPostScheduled(),
+		};
+	} );
```
Comment (n3f) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/settings.js:74]: This made me scratch my head for a second.  Looks good.

**Review by n3f — DISMISSED** [review]
LGTM

**Review by fgiannar — DISMISSED** [review]
LGTM 👍

**Comment by romarioraffington** [comment]
@fgiannar  I had some issues rebasing and had to force push. The branch is now updated with trunk. Another 👍  would be appreciated!

**Review by zinigor — APPROVED** [review]

**Pull request merged**